src/common/get_messages_in_datasets.py

```diff
@@ -124,4 +127,31 @@ def get_messages_in_datasets(engagement_db, engagement_db_datasets, cache=None,
                                                                     f"'{msg.origin.origin_id}'"
             all_message_origins.add(msg.origin.origin_id)
 
+    # Filter out messages that don't meet the status conditions
+    for engagement_db_dataset, messages in engagement_db_messages_map.items():
+        # Filter out messages that do not have status "live" or "stale"
+        active_messages = [msg for msg in messages if msg.status in {MessageStatuses.LIVE, MessageStatuses.STALE}]
```

Comment:
The line where we append stale messages to live messages makes the code a little bit confusing. I think it's the active messages that should be appended to i.e, L143. I think we should introduce another variable name that is related to active messages, in this case, `actual_active_messages`.

I've added a few suggestions here and below to clearly articulate how I think this can be achieved.
```
active_messages = defaultdict(list)
active_messages["live_messages"].extend([msg for msg in messages if msg.status == "live"])
active_messages["stale_messages"].extend([msg for msg in messages if msg.status == "stale"])
```

Reply:
Thanks, I think this much clearer, so have reworked this along the lines suggested here